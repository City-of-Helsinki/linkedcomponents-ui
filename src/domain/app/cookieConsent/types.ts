export type Consent = {
  acceptedAt: string;
  required: boolean;
  tracking: boolean;
};
