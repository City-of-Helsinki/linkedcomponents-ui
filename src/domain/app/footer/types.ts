export type NavigationGroup = {
  heading: string;
  headingLink: string;
  items?: NavigationGroupItem[];
};

export type NavigationGroupItem = {
  label: string;
  url: string;
};
