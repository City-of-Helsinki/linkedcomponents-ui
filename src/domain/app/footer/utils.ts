import { ROUTES } from '../../../constants';
import { UserFieldsFragment } from '../../../generated/graphql';
import { areRegistrationRoutesAllowed } from '../../user/permissions';
import { NavigationGroup, NavigationGroupItem } from './types';

const getNavigationGroupItemsInstructions = (
  user?: UserFieldsFragment
): NavigationGroupItem[] => [
  {
    label: 'helpPage.sideNavigation.labelEventsInstructions',
    url: ROUTES.INSTRUCTIONS_EVENTS,
  },
  ...(areRegistrationRoutesAllowed(user)
    ? [
        {
          label: 'helpPage.sideNavigation.labelRegistrationInstructions',
          url: ROUTES.INSTRUCTIONS_REGISTRATION,
        },
      ]
    : []),
  {
    label: 'helpPage.sideNavigation.labelFaq',
    url: ROUTES.INSTRUCTIONS_FAQ,
  },
];

export const getNavigationGroupInstructions = (
  user?: UserFieldsFragment
): NavigationGroup => ({
  heading: 'helpPage.sideNavigation.labelInstructions',
  headingLink: ROUTES.INSTRUCTIONS,
  items: getNavigationGroupItemsInstructions(user),
});
