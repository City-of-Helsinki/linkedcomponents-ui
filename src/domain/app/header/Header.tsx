import { css } from '@emotion/css';
import classNames from 'classnames';
import { IconInfoCircle, IconSignout, Navigation } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { matchPath, RouteProps, useHistory, useLocation } from 'react-router';

import { MAIN_CONTENT_ID, PAGE_HEADER_ID, ROUTES } from '../../../constants';
import useLocale from '../../../hooks/useLocale';
import useSelectLanguage from '../../../hooks/useSelectLanguage';
import { isFeatureEnabled } from '../../../utils/featureFlags';
import { signIn, signOut } from '../../auth/authenticate';
import { authenticatedSelector } from '../../auth/selectors';
import { getEventSearchQuery } from '../../eventSearch/utils';
import useUser from '../../user/hooks/useUser';
import { useTheme } from '../theme/Theme';
import styles from './header.module.scss';

interface NoNavRowProps {
  pathname: string;
  props?: RouteProps;
}

interface NavigationItem {
  className?: string;
  icon?: React.ReactElement;
  labelKey: string;
  url: ROUTES;
}

const NO_NAV_ROW_PATHS = [
  { pathname: ROUTES.EDIT_EVENT },
  { pathname: ROUTES.EDIT_REGISTRATION },
  { pathname: ROUTES.EDIT_REGISTRATION_ENROLMENT },
  { pathname: ROUTES.REGISTRATION_ENROLMENTS },
];

const SCROLL_OFFSET = 40;

const Header: React.FC = () => {
  const { theme } = useTheme();
  const locale = useLocale();
  const history = useHistory();
  const location = useLocation();
  const { changeLanguage, languageOptions } = useSelectLanguage();

  const { t } = useTranslation();
  const authenticated = useSelector(authenticatedSelector);
  const user = useUser();
  const [menuOpen, setMenuOpen] = React.useState(false);

  const isTabActive = (pathname: string): boolean => {
    return location.pathname.startsWith(pathname);
  };

  const goToHomePage = (e?: Event) => {
    e?.preventDefault();
    history.push({ pathname: `/${locale}${ROUTES.HOME}` });
    setMenuOpen(false);
  };

  const goToPage =
    (pathname: string) => (e?: React.MouseEvent<HTMLAnchorElement>) => {
      e?.preventDefault();
      history.push({ pathname });
      toggleMenu();
    };

  const NAVIGATION_ITEMS = [
    { labelKey: 'navigation.tabs.events', url: ROUTES.EVENTS },
    isFeatureEnabled('SHOW_REGISTRATION') && {
      labelKey: 'navigation.tabs.registrations',
      url: ROUTES.REGISTRATIONS,
    },
    isFeatureEnabled('SHOW_ADMIN') && {
      labelKey: 'navigation.tabs.admin',
      url: ROUTES.ADMIN,
    },
    {
      className: styles.navigationItemLast,
      icon: <IconInfoCircle aria-hidden={true} />,
      labelKey: 'navigation.tabs.help',
      url: ROUTES.HELP,
    },
  ].filter((i) => i) as NavigationItem[];

  const navigationItems = NAVIGATION_ITEMS.map(
    ({ labelKey, url, ...rest }) => ({
      label: t(labelKey),
      url: `/${locale}${url}`,
      ...rest,
    })
  );

  const handleSignIn = () => {
    signIn(`${location.pathname}${location.search}`);
  };

  const handleSignOut = (e: React.MouseEvent) => {
    e.preventDefault();
    signOut();
  };

  const isMatch = (paths: NoNavRowProps[]) =>
    paths.some((path) =>
      matchPath(location.pathname, {
        path: `/${locale}${path.pathname}`,
        exact: path.props?.exact ?? true,
        strict: path.props?.strict ?? true,
      })
    );

  const noNavRow = isMatch(NO_NAV_ROW_PATHS);

  const toggleMenu = () => {
    setMenuOpen(!menuOpen);
  };

  const handleSearch = (text: string) => {
    history.push({
      pathname: `/${locale}${ROUTES.SEARCH}`,
      search: getEventSearchQuery({ text }),
    });
    toggleMenu();
  };

  /* istanbul ignore next */
  const onDocumentFocusin = (event: FocusEvent) => {
    const target = event.target;
    const navigation = document.querySelector(`#${PAGE_HEADER_ID}`);

    if (
      target instanceof HTMLElement &&
      navigation instanceof HTMLElement &&
      !navigation.contains(target)
    ) {
      const navigationRect = navigation.getBoundingClientRect();
      const targetRect = target.getBoundingClientRect();

      if (navigationRect && navigationRect.bottom > targetRect.top) {
        window.scrollBy(
          0,
          targetRect.top - navigationRect.bottom - SCROLL_OFFSET
        );
      }
    }
  };

  React.useEffect(() => {
    document.addEventListener('focusin', onDocumentFocusin);

    return () => {
      document.removeEventListener('focusin', onDocumentFocusin);
    };
  });

  return (
    <Navigation
      id={PAGE_HEADER_ID}
      menuOpen={menuOpen}
      onMenuToggle={toggleMenu}
      menuToggleAriaLabel={t('navigation.menuToggleAriaLabel')}
      skipTo={`#${MAIN_CONTENT_ID}`}
      skipToContentLabel={t('navigation.skipToContentLabel')}
      className={classNames(css(theme.navigation), styles.navigation, {
        [styles.hideNavRow]: noNavRow,
      })}
      onTitleClick={goToHomePage}
      title={t('appName')}
      titleUrl={`/${locale}${ROUTES.HOME}`}
      logoLanguage={locale === 'sv' ? /* istanbul ignore next */ 'sv' : 'fi'}
    >
      <Navigation.Row>
        {navigationItems.map((item, index) => (
          <Navigation.Item
            key={index}
            active={isTabActive(item.url)}
            className={classNames(styles.navigationItem, item.className)}
            icon={item.icon}
            href={item.url}
            label={item.label}
            onClick={goToPage(item.url)}
          />
        ))}
      </Navigation.Row>
      <Navigation.Actions>
        <Navigation.Search
          onSearch={handleSearch}
          searchLabel={t('navigation.searchEvents')}
        />
        {/* USER */}
        <Navigation.User
          authenticated={Boolean(authenticated && user.user)}
          className={classNames(
            styles.userDropdown,
            css(theme.navigationDropdown)
          )}
          label={t('common.signIn')}
          onSignIn={handleSignIn}
          userName={user?.user?.displayName || user?.user?.email}
        >
          <Navigation.Item
            label={t('common.signOut')}
            href="#"
            icon={<IconSignout aria-hidden />}
            variant="supplementary"
            onClick={handleSignOut}
          />
        </Navigation.User>
        <Navigation.LanguageSelector
          buttonAriaLabel={t('navigation.languageSelectorAriaLabel')}
          className={classNames(
            styles.languageSelector,
            css(theme.navigationDropdown)
          )}
          label={t(`navigation.languages.${locale}`)}
        >
          {languageOptions.map((option) => (
            <Navigation.Item
              key={option.value}
              href="#"
              lang={option.value}
              label={option.label}
              onClick={changeLanguage(option)}
            />
          ))}
        </Navigation.LanguageSelector>
      </Navigation.Actions>
    </Navigation>
  );
};

export default Header;
