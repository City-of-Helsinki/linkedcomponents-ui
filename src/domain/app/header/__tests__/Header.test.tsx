import { AnyAction, Store } from '@reduxjs/toolkit';
import i18n from 'i18next';
import React from 'react';

import { ROUTES } from '../../../../constants';
import { StoreState } from '../../../../types';
import { fakeAuthenticatedStoreState } from '../../../../utils/mockStoreUtils';
import {
  getMockReduxStore,
  render,
  screen,
  userEvent,
  waitFor,
} from '../../../../utils/testUtils';
import translations from '../../../app/i18n/fi.json';
import userManager from '../../../auth/userManager';
import Header from '../Header';

const renderComponent = (store?: Store<StoreState, AnyAction>, route = '/fi') =>
  render(<Header />, { routes: [route], store });

const findComponent = (
  key:
    | 'languageSelector'
    | 'menuButton'
    | 'navigation'
    | 'signInButton'
    | 'signOutLink'
    | 'svOption'
) => {
  switch (key) {
    case 'languageSelector':
      return screen.findByRole('button', {
        name: translations.navigation.languageSelectorAriaLabel,
      });
    case 'menuButton':
      return screen.findByRole('button', {
        name: translations.navigation.menuToggleAriaLabel,
      });
    case 'navigation':
      return screen.findByRole('navigation');
    case 'signInButton':
      return screen.findByRole('button', { name: translations.common.signIn });
    case 'signOutLink':
      return screen.findByRole('link', { name: translations.common.signOut });
    case 'svOption':
      return screen.findByRole('link', {
        name: translations.navigation.languages.sv,
      });
  }
};

beforeEach(() => {
  jest.restoreAllMocks();
  i18n.changeLanguage('fi');
});

test('matches snapshot', async () => {
  i18n.changeLanguage('sv');
  const { container } = renderComponent(undefined, '/sv');

  expect(container.firstChild).toMatchSnapshot();
});

test('should show navigation links and should route to correct page after clicking link', async () => {
  const { history } = renderComponent();
  const links = [
    {
      name: translations.navigation.tabs.events,
      url: `/fi${ROUTES.EVENTS}`,
    },
    {
      name: translations.navigation.tabs.searchEvent,
      url: `/fi${ROUTES.SEARCH}`,
    },
    {
      name: translations.navigation.tabs.help,
      url: `/fi${ROUTES.HELP}`,
    },
  ];

  links.forEach(({ name, url }) => {
    const link = screen.getByRole('link', { name });

    expect(link).toBeInTheDocument();

    userEvent.click(link);

    expect(history.location.pathname).toBe(url);
  });
});

test('should show mobile menu', async () => {
  global.innerWidth = 500;
  renderComponent();

  expect(screen.queryByRole('navigation')).not.toBeInTheDocument();

  const menuButton = await findComponent('menuButton');
  userEvent.click(menuButton);

  await findComponent('navigation');
});

test('should change language', async () => {
  global.innerWidth = 1200;
  const { history } = renderComponent();

  expect(history.location.pathname).toBe('/fi');

  const languageSelectorButton = await findComponent('languageSelector');
  userEvent.click(languageSelectorButton);

  const svOption = await findComponent('svOption');
  userEvent.click(svOption);

  expect(history.location.pathname).toBe('/sv');
});

test('should start log in process', async () => {
  const signinRedirect = jest.spyOn(userManager, 'signinRedirect');

  renderComponent();

  const signInButton = await findComponent('signInButton');
  userEvent.click(signInButton);

  expect(signinRedirect).toBeCalled();
});

test('should start logout process', async () => {
  const signoutRedirect = jest.spyOn(userManager, 'signoutRedirect');

  const storeState = fakeAuthenticatedStoreState();
  const userName = storeState.authentication.oidc.user.profile.name;
  const store = getMockReduxStore(storeState);

  renderComponent(store);

  const userMenuButton = await screen.findByRole('button', { name: userName });
  userEvent.click(userMenuButton);

  const signOutLink = await findComponent('signOutLink');
  userEvent.click(signOutLink);

  await waitFor(() => {
    expect(signoutRedirect).toBeCalled();
  });
});
