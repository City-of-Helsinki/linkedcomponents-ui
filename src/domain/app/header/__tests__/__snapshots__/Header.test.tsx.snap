// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<header
  class="hds-header Header-module_header__291GF navigation custom-theme-1"
  id="page-header"
  theme="[object Object]"
>
  <div
    class="Header-module_headerBackgroundWrapper__KsFy2"
  >
    <a
      class="SkipLink-module_skipLink__1knzF"
      href="#maincontent"
    >
      <span
        class="SkipLink-module_skipLinkLabel__21n5B"
      >
        Hoppa till innehåll
      </span>
    </a>
    <div
      class="HeaderActionBar-module_headerActionBarContainer__5tfj-"
    >
      <div
        class="HeaderActionBar-module_headerActionBar__lPeGE"
      >
        <a
          class="HeaderActionBar-module_titleAndLogoContainer__2LEAb HeaderActionBar-module_logo__2r2T0"
          href="/sv/"
          role="link"
        >
          <span
            class="HeaderActionBarLogo-module_logoWrapper__8NgfS"
          >
            <img
              alt="Helsingfors stad - logo"
              class="Logo-module_logo__Y2mwP Logo-module_medium__1zyWm"
              src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0idXRmLTgiPz4KPCFET0NUWVBFIHN2Zz4KPHN2ZyB2aWV3Qm94PSIwIDAgMTA0IDM2IiB0aXRsZT0iSGVsc2luZ2ZvcnMgc3RhZCIgcm9sZT0iaW1nIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciCiAgICBjbGFzcz0iTG9nby1tb2R1bGVfbG9nb19fWTJtd1AgTG9nby1tb2R1bGVfbWVkaXVtX18xenlXbSIgc3R5bGU9Im1hcmdpbi1yaWdodDogdmFyKC0tc3BhY2luZy14cyk7Ij4KICAgIDxwYXRoCiAgICAgICAgZD0iTTEwMy4xMDUgMHYyMi45MzVjMCA1LjE3OS00LjI2NiA5LjM5Ny05LjUyNCA5LjM5N2wtMzQuODY3LjAzMmE4LjQ2IDguNDYgMCAwMC01Ljk4MSAyLjQ3MUw1MS41NTMgMzZsLTEuMTgxLTEuMTY1YTguNDA1IDguNDA1IDAgMDAtNS45ODItMi40NzFIOS40NzZDNC4yNSAzMi4zNjQgMCAyOC4xNDUgMCAyMi45NjZWMGgxMDMuMTA1em0tMi4yMzYgMi4yMzVIMi4yNTF2MjAuNzMxYzAgMy45NTEgMy4yNDMgNy4xNjMgNy4yMjUgNy4xNjNINDQuMzljMi42NzYgMCA1LjE5NS45NiA3LjE2MiAyLjczOSAxLjk2OC0xLjc4IDQuNDg2LTIuNzQgNy4xNjItMi43NGwzNC44NjctLjAzYzQuMDE0IDAgNy4yODgtMy4yMTIgNy4yODgtNy4xNjN2LTIwLjd6TTYyLjAyIDEyLjA5bDIgMS4xNWMtLjAzMiAxLjAwNy0uNDg5IDEuNDc5LTEuMzU0IDEuNDc5LS4yMDUgMC0uNDczLS4wNDctLjYxNC0uMDk0di4wNjJjLjQyNS4zMzEuOTQ0Ljg4Mi45NDQgMS43NjMgMCAxLjUxMi0xLjQ2NCAyLjQ3Mi0zLjI3NCAyLjQ3Mi0uODM0IDAtMS41NTguMjY3LTEuNTU4LjY3NyAwIC4zMTUuMjY3LjQ1Ni45MTMuNDU2LjY0NSAwIDEuMzY5LS4xMjYgMi4yMDMtLjEyNiAxLjY1MyAwIDIuNzU1LjY2MSAyLjc1NSAyLjE1NyAwIDEuNzk0LTIuMDE1IDIuNTAzLTQuMjgyIDIuNTAzLTIuMTA5IDAtNC4wMy0uNTk4LTQuMDMtMS45NTIgMC0uODE5Ljg1LTEuMzA3IDEuNjIyLTEuNDQ4di0uMDQ4Yy0uNTgyLS4xNzMtMS4xMDItLjU1LTEuMTAyLTEuMjEyIDAtLjg2NS44ODItMS4zMzggMS43NzktMS41Mjd2LS4wNDdjLS44MzQtLjMzLTEuNTQzLTEuMDA3LTEuNTQzLTIuMTg4IDAtMS41MTEgMS4zMjItMi41NSAzLjEzMy0yLjU1LjY5MiAwIDEuMDg2LjE0MiAxLjYwNS4xNDIuNjQ2IDAgLjg4Mi0uMzYyLjg4Mi0uOTQ1IDAtLjI2Ny0uMDQ3LS41MzUtLjA3OS0uNzI0em0tMS43MzEgOS41NGMtLjkzIDAtMi43MjQuMDc4LTIuNzI0Ljc4NiAwIC41MzYuODk4Ljc4NyAyLjE1Ny44MDMgMS40MTcgMCAyLjIzNS0uMjY3IDIuMjM1LS45MTMgMC0uNTA0LS43NTUtLjY3Ny0xLjY2OC0uNjc3em0yOC45OTUtOC4wMTNjMS41MjcgMCAyLjkyOC43MjQgMy41MjYgMS43NDdsLTEuODQyIDEuMDRhMS42MjQgMS42MjQgMCAwMC0xLjYwNi0xLjIxM2MtLjU1IDAtMS4wMDcuMjM2LTEuMDA3LjY3NyAwIC41ODIgMS4wMDcuNjQ1IDIuMTU3IDEuMDA4IDEuMTk2LjM3NyAyLjI5OC45MjggMi4yOTggMi4zMyAwIDEuNTQyLTEuNDMzIDIuNDU1LTMuMTAxIDIuNDU1LTEuNzk1IDAtMy4yMTEtLjc3MS0zLjg0MS0xLjk4NGwxLjg3My0xLjA1NGMuMjUyLjgzNC45MTMgMS40NjQgMS45MzYgMS40NjQuNTk4IDAgMS4wMjMtLjI2OCAxLjAyMy0uNzQgMC0uNjE0LS44NS0uNzU2LTEuODQxLTEuMDIzLTEuMTY1LS4zMTUtMi42MTMtLjg4Mi0yLjYxMy0yLjM0NiAwLTEuNDQ4IDEuMzg1LTIuMzYgMy4wMzgtMi4zNnptLTE0LjY1NSAwYzIuMDc4IDAgMy42OTkgMS41MjcgMy42OTkgNC4wMTRzLTEuNjA2IDQuMDE0LTMuNyA0LjAxNGMtMi4xMDkgMC0zLjcxNC0xLjUyNy0zLjcxNC00LjAxNHMxLjYwNS00LjAxNCAzLjcxNS00LjAxNHptLTQ5LjY4IDBjMi4wNjMgMCAzLjQzMiAxLjQ2NCAzLjQzMiAzLjU0Mi4wMTYuNjE0LS4wOTQgMS4wMzktLjA5NCAxLjAzOWgtNC45NzRjLjA5NCAxLjI0My43ODcgMS44NTcgMS42ODQgMS44NTcuODAzIDAgMS4zNTQtLjUzNSAxLjQ4LTEuMjZsMS44MSAxLjAyNGMtLjUzNS45NzYtMS42ODQgMS44MjYtMy4yOSAxLjgyNi0yLjE0IDAtMy43MTUtMS40OTUtMy43MTUtNC4wMTQgMC0yLjQ4NyAxLjYwNi00LjAxNCAzLjY2OC00LjAxNHptMTMuMDY2LS4wMTZjMS41MjcgMCAyLjkyOC43MjQgMy41MjYgMS43NDhsLTEuODQyIDEuMDM5Yy0uMTg5LS42NzctLjc1NS0xLjIxMy0xLjYwNS0xLjIxMy0uNTUxIDAtMS4wMDguMjM3LTEuMDA4LjY3NyAwIC41ODMgMS4wMDguNjQ2IDIuMTQgMS4wMDggMS4xOTcuMzc4IDIuMy45MjggMi4zIDIuMzMgMCAxLjU0Mi0xLjQzMyAyLjQ0LTMuMDg2IDIuNDQtMS43OTUgMC0zLjIxMS0uNzU2LTMuODQxLTEuOTg0bDEuODczLTEuMDU1Yy4yMzYuODM1LjkxMyAxLjQ2NCAxLjkzNiAxLjQ2NC41ODMuMDE2IDEuMDA4LS4yMzYgMS4wMDgtLjcyNCAwLS42MTQtLjg1LS43NC0xLjg0Mi0xLjAyMy0xLjE2NS0uMy0yLjU5Ny0uODgyLTIuNTk3LTIuMzQ1IDAtMS40NDkgMS4zODUtMi4zNjIgMy4wMzgtMi4zNjJ6TTMyLjMgMTAuNTN2OC41MDFjMCAuMzYyLjA0Ny42MTQuMTU3Ljc1Ni4wOTUuMTU3LjIyLjIyLjUwNC4yMi4xNzMgMCAuMzMgMCAuNDU2LS4wMzFhMi4wMSAyLjAxIDAgMDAuNDU3LS4xMjZsLS4xODkgMS40NDhhMi41MzIgMi41MzIgMCAwMS0uNjQ1LjIzNmMtLjI1Mi4wNDctLjUyLjA3OS0uNzcyLjA3OS0uNzQgMC0xLjI3NS0uMTc0LTEuNjIxLS41MzYtLjMzLS4zNjItLjUwNC0uOTQ0LS41MDQtMS43MzFWMTAuNTNoMi4xNTd6bTE5LjgxOCAzLjA4NmMxLjQ4IDAgMi40MDggMS4wMjMgMi40MDggMi45Mjh2NC45MTFoLTIuMTU2VjE2Ljk3YzAtMS4wMDctLjM2Mi0xLjU0My0xLjE2NS0xLjU0M3MtMS4zMjIuNjE0LTEuMzIyIDEuNDk2djQuNTQ5aC0yLjE0MXYtNy42NjZoMi4xNGwtLjExIDEuMjZoLjA0OGMuMzc4LS43MSAxLjAzOS0xLjQ0OSAyLjI5OC0xLjQ0OXptMTYuNzMzLTMuMzM3YTMuMjcgMy4yNyAwIDAxMi43MzkgMS40bC0xLjg1OCAxLjEzNGMtLjAzMS0uNDU2LS4yNTItLjkyOC0uNzcxLS45MjgtLjUzNSAwLS43ODcuNDA5LS43ODcgMS4wODZ2LjgzNGgxLjY4NHYxLjYzN2gtMS42ODR2Ni4wMjloLTIuMTR2LTYuMDI5aC0xLjE4MXYtMS42MzdoMS4xOHYtLjgxOWMwLTEuNzk0IDEuMTgtMi43MDcgMi44MTgtMi43MDd6bTE0LjgxMiAzLjM4NGMxLjA4NiAwIDEuNjg0LjkxMyAxLjY4NCAyLjAzIDAgLjUwNS0uMDc4Ljg5OC0uMDc4Ljg5OGwtMS45MzYgMS4xMThjLjAzMS0uMjIuMTI2LS43NC4xMjYtMS4yNiAwLS41MDMtLjE5LS45NzUtLjcyNS0uOTc1LS42NzYgMC0uOTQ0LjYzLS45NDQgMS42MnY0LjM3N2gtMi4xNTd2LTcuNjVoMi4xNTdsLS4xMSAxLjI1OWguMDQ3Yy4zNzgtLjc0Ljk2LTEuNDE3IDEuOTM2LTEuNDE3em0tNzAuNTItMi43MjN2NC4xNzFoNC4yMDN2LTQuMTdoMi4yMzV2MTAuNTE0aC0yLjIzNXYtNC4zNmgtNC4yMDN2NC4zNmgtMi4yMzVWMTAuOTRoMi4yMzV6bTMyLjMzMiAyLjg1djcuNjY1SDQzLjMyVjEzLjc5aDIuMTU2em0yOS4xNTMgMS40NjNjLTEuMDcgMC0xLjU0MyAxLjAwOC0xLjU0MyAyLjM3NyAwIDEuMzcuNDcyIDIuMzc3IDEuNTQzIDIuMzc3IDEuMDU0IDAgMS41MjctMS4wMDcgMS41MjctMi4zNzcgMC0xLjM3LS40NzMtMi4zNzctMS41MjctMi4zNzd6bS0xNC44OTEtLjE4OWMtLjcyNCAwLTEuMzA3LjQ3My0xLjMwNyAxLjE5NyAwIC43NC41ODMgMS4yMTIgMS4zMDcgMS4xOTYuNzI0IDAgMS4zMDYtLjQ3MiAxLjMwNi0xLjE5NnMtLjU4Mi0xLjE5Ny0xLjMwNi0xLjE5N3ptLTM0Ljc3My4wOTVjLS43ODcgMC0xLjQzMi41NjYtMS42MDUgMS42MDVoMy4wMzhjMC0uOTEzLS41OTgtMS42MDUtMS40MzMtMS42MDV6bTE5LjQ0LTQuNjc1Yy42OTMgMCAxLjI0NC41MDMgMS4yNDQgMS4xOTZzLS41NSAxLjE5Ni0xLjI0MyAxLjE5NmMtLjcwOSAwLTEuMjQ0LS41MDMtMS4yNDQtMS4xOTZzLjU1MS0xLjE5NiAxLjI0NC0xLjE5NnoiCiAgICAgICAgZmlsbD0id2hpdGUiPjwvcGF0aD4KPC9zdmc+"
            />
          </span>
        </a>
        <a
          class="HeaderActionBar-module_titleAndLogoContainer__2LEAb HeaderActionBar-module_title__opzt9 HeaderActionBar-module_normal__3B4qi"
          href="/sv/"
        >
          <span
            class="HeaderActionBar-module_title__opzt9"
          >
            Linked Events
          </span>
        </a>
        <div
          class="HeaderActionBar-module_headerActions__3lNnM"
        >
          <div
            class="HeaderLanguageSelector-module_languageSelector__3d38B"
          >
            <div
              class="HeaderLanguageSelector-module_languageNodes__lTe8p"
            >
              <button
                aria-current="false"
                class="HeaderLanguageSelector-module_item__3_I7N"
                lang="fi"
                type="button"
              >
                <span>
                  Suomeksi
                </span>
              </button>
              <button
                aria-current="true"
                class="HeaderLanguageSelector-module_item__3_I7N HeaderLanguageSelector-module_activeItem__15LKg"
                lang="sv"
                type="button"
              >
                <span>
                  På svenska
                </span>
              </button>
              <button
                aria-current="false"
                class="HeaderLanguageSelector-module_item__3_I7N"
                lang="en"
                type="button"
              >
                <span>
                  In English
                </span>
              </button>
            </div>
          </div>
          <hr
            aria-hidden="true"
          />
          <div
            class="HeaderActionBarItem-module_container__1uq2m HeaderActionBarItem-module_hasContent__YJfOU HeaderActionBarItem-module_hasSubItems__1LVkh"
            id="action-bar-user"
          >
            <button
              aria-controls="action-bar-user-dropdown"
              aria-expanded="false"
              aria-label="Test user"
              class="HeaderActionBarItemButton-module_actionBarItemButton__2ALyH HeaderActionBarItem-module_icon__1Ug_k HeaderActionBarItemButton-module_fixedRightPosition__kBVLv HeaderActionBarItemButton-module_hasSubItems__3wX3Y"
              type="button"
            >
              <div
                class="HeaderActionBarItemButton-module_buttonContent__3JeY0"
              >
                <span>
                  <span
                    class="HeaderActionBarItemButton-module_actionBarItemButtonIcon__1_1Cg"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="user"
                      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_small__gTGkU icon_hds-icon--size-s__2Lkik"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.2667 13C18.8082 13 20.8819 15.0031 20.9951 17.5167L21 17.7333V22H3V17.7333C3 15.1918 5.00309 13.1181 7.51667 13.0049L7.73333 13H16.2667ZM7.73333 15C6.27966 15 5.09104 16.1348 5.00499 17.5668L5 17.7333V20H19V17.7333C19 16.3172 17.8926 15.0973 16.54 15.0075L16.3766 15.0022L7.73333 15ZM12 2C14.7614 2 17 4.23858 17 7C17 9.76142 14.7614 12 12 12C9.23858 12 7 9.76142 7 7C7 4.23858 9.23858 2 12 2ZM12 4C10.3431 4 9 5.34315 9 7C9 8.65685 10.3431 10 12 10C13.6569 10 15 8.65685 15 7C15 5.34315 13.6569 4 12 4Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <span
                    class="HeaderActionBarItemButton-module_actionBarItemButtonLabel__12Iec"
                  >
                    Test user
                  </span>
                </span>
                <svg
                  aria-hidden="true"
                  aria-label="angle-down"
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_small__gTGkU icon_hds-icon--size-s__2Lkik"
                  role="img"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M12 13.5L17 8.5L18.5 10L12 16.5L5.5 10L7 8.5L12 13.5Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </button>
            <div
              class="HeaderActionBarItem-module_dropdownWrapper__39cBw"
            >
              <div
                class="HeaderActionBarItem-module_dropdown__3daDZ HeaderActionBarItem-module_hasContent__YJfOU HeaderActionBarItem-module_hasSubItems__1LVkh"
                id="action-bar-user-dropdown"
              >
                <ul>
                  <a
                    class="Link-module_link__TeBQo link_hds-link__5Oxo- Link-module_link-medium__3gvla link_hds-link--medium__xEU_F signoutLink"
                    href="#"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="signout"
                      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_small__gTGkU icon_hds-icon--size-s__2Lkik"
                      role="img"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16 2V7H14V4H5V20H14V17H16V22H3V2H16ZM18 7L23 12L18 17L16.5 15.5L19 12.999L8 13V11L18.999 10.999L16.5 8.5L18 7Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                    Logga ut
                  </a>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="HeaderNavigationMenu-module_headerNavigationMenuContainer__2nK9z"
    >
      <nav
        class="HeaderNavigationMenu-module_headerNavigationMenu__3G6-N"
      >
        <ul
          class="HeaderNavigationMenu-module_headerNavigationMenuList__3HIzx"
        >
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="Link-module_link__TeBQo link_hds-link__5Oxo- Link-module_link-medium__3gvla link_hds-link--medium__xEU_F HeaderLink-module_headerLink__2i9-J navigationItem HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/sv/events"
                >
                  Mina evenemangen
                </a>
              </span>
            </span>
          </li>
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="Link-module_link__TeBQo link_hds-link__5Oxo- Link-module_link-medium__3gvla link_hds-link--medium__xEU_F HeaderLink-module_headerLink__2i9-J navigationItem HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/sv/search"
                >
                  Sök efter evenemang
                </a>
              </span>
            </span>
          </li>
          <li>
            <span
              class="HeaderNavigationMenu-module_headerNavigationMenuLinkContent__3zC2j"
            >
              <span
                class="HeaderLink-module_navigationLinkWrapper__2jd7A"
              >
                <a
                  class="Link-module_link__TeBQo link_hds-link__5Oxo- Link-module_link-medium__3gvla link_hds-link--medium__xEU_F HeaderLink-module_headerLink__2i9-J navigationItem HeaderNavigationMenu-module_headerNavigationMenuLink__H_ogS"
                  href="/sv/help"
                >
                  Stöd
                </a>
              </span>
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
